services/audit/appender.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package audit
+
+import (
+	"context"
+	"io"
+
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/util/rotatingfilewriter"
+)
+
+type Appender interface {
+	Record(context.Context, *Event)
+	Close() error
+	ReleaseReopen() error
+}
+
+// LogAppender writes an info log entry for every audit event
+type LogAppender struct{}
+
+func (a *LogAppender) Record(ctx context.Context, e *Event) {
+	log.Info(e.Message)
```

Comment:
What is this "log appender" used for? The `Message` is from `fmt.Sprintf(format, v...)`, then all other fields in the `Event` is lost?

Reply:
Yes, it just logs the message to the normal log. I don't think someone wants more info in that case. Otherwise you would choose the file logging. See this appender as a "does audit logging work" check.

Reply:
> See this appender as a "does audit logging work" check.

Do you mean it is for test purpose only? If yes, it could be moved to the test-related package.

Reply:
No, someone may just have the audit logs integrated into normal logging.

Reply:
Who? The message contains barely nothing, no scope, no target, why they would like to see it in normal logging?

Reply:
Because they do not need/want the full json output. So you want more context in the build message? Currently the message is just a quick readable info what this log entry is about. This type of appender is not relevant for corporate audit logging. It's intended for small instances.

Reply:
Well, I do not feel it is good, or it makes sense for "small instances".

But if you are sure there are such users, I don't block.

Reply:
How about this? The message followed by scope and target:
```
[I] Created repository KN4CK3R/audit-test. [repository: 67, KN4CK3R/audit-test] [repository: 67, KN4CK3R/audit-test]
[I] Changed settings of repository KN4CK3R/audit-test. [repository: 67, KN4CK3R/audit-test] [repository: 67, KN4CK3R/audit-test]
[I] Changed visibility of repository KN4CK3R/audit-test to private. [repository: 67, KN4CK3R/audit-test] [repository: 67, KN4CK3R/audit-test]
[I] Added user te.st as collaborator for repository KN4CK3R/audit-test. [repository: 67, KN4CK3R/audit-test] [user: 15, te.st]
```

Reply:
Yup, I guess that's the "Structured logger" pattern (like the discussion above)